frontend/src/modules/history/components/Translation.js

```diff
@@ -168,6 +168,78 @@ export class TranslationBase extends React.Component<InternalProps, State> {
         </Localized>
     }
 
+    renderCommentSummary() {
+        const comments = this.props.translation.comments;
+        const commentCount = comments.length;
+
+        if (commentCount === 0) {
+            return null;
+        }
+
+        const authors = [...new Set(comments.map(comment => comment.username))];
```

Comment:
You could return an object in your `map` call here, including both `username` and `userGravatarUrlSmall`, then not need to do another recursion below when displaying user avatars.